data/transactions/logic/assembler.go

```diff
@@ -1531,6 +1534,38 @@ func (ops *OpStream) trackStack(args StackTypes, returns StackTypes, instruction
 	}
 }
 
+func processFields(ops *OpStream, fields []string) (current, next []string) {
+	for i := 0; i < len(fields); i++ {
+		field := fields[i]
+		replacement, ok := ops.macros[field]
+		if ok {
+			fields = append(fields[0:i], append(replacement, fields[i+1:]...)...)
+			i--
+			continue
+		}
+		if string(field[len(field)-1]) == ";" {
+			field = field[0 : len(field)-1]
+			replacement, ok = ops.macros[field]
+			if ok {
+				last := replacement[len(replacement)-1]
+				fields = append(fields[0:i], append(replacement, fields[i+1:]...)...)
+				fields[i+len(replacement)-1] = last + ";"
+				i--
+				continue
+			}
+			current = fields[:i]
+			if len(field) > 0 {
+				current = append(current, field)
+			}
```

Comment:
This modifies `fields`! When you take a slice of `fields`, and then append to it, it shares the underlying array. If there's room, the append writes into it.

Reply:
Ya but it only writes to fields[i] and we only care about fields[i+1:] from that point I think